test/cqlpy/conftest.py

```diff
@@ -230,15 +231,19 @@ def random_seed():
 # executable's path. If we can't find the Scylla executable we use
 # pytest.skip() to skip tests relying on this executable.
 @pytest.fixture(scope=testpy_test_fixture_scope)
-def scylla_path(cql):
-    pid = local_process_id(cql)
-    if not pid:
-        pytest.skip("Can't find local Scylla process")
-    # Now that we know the process id, use /proc to find the executable.
-    try:
-        path = os.readlink(f'/proc/{pid}/exe')
-    except:
+def scylla_path(cql, build_mode):
+    if TEST_RUNNER == "runpy":
+        pid = local_process_id(cql)
+        if not pid:
+            pytest.skip("Can't find local Scylla process")
+        # Now that we know the process id, use /proc to find the executable.
+        try:
+            path = os.readlink(f'/proc/{pid}/exe')
+        except:
+            print(f"/proc/{pid}/exe doesn't exist")
```

Comment:
The commit message doesn't really explain what's going on here... "tests
are just randomly skipped"? When? Why? Why does xdist prevent this code from finding the Scylla executable - shouldn't this code work just fine even with xdist?

The TEST_RUNNER trick may work, so I'm not against it, but I stilll don't like it.  I think it encourages bad assumptions and its existence makes it seem like it's a strange request to run tests against an unknown versions of Scylla (which may be old versions of even Cassandra) with no "build mode" or anything like that. While I consider this ability to run against non-standard Scyllas a super fundemental requirement from the test framework (while sadly it was never a requirement of test.py).